SimG4CMS/Calo/src/HFShowerLibrary.cc

```diff
@@ -335,223 +397,299 @@ std::vector<HFShowerLibrary::Hit> HFShowerLibrary::fillHits(const G4ThreeVector&
   }
 
 #ifdef EDM_ML_DEBUG
-  edm::LogVerbatim("HFShower") << "HFShowerLibrary: Total Hits " << nHit << " out of " << npe << " PE";
+  edm::LogVerbatim("HFShower") << "HFShowerLibrary: Total Hits " << nHit << " out of " << pe.size() << " PE";
 #endif
-  if (nHit > npe && !onlyLong) {
-    edm::LogWarning("HFShower") << "HFShowerLibrary: Hit buffer " << npe << " smaller than " << nHit << " Hits";
+  if (nHit > pe.size() && !onlyLong) {
+    edm::LogWarning("HFShower") << "HFShowerLibrary: Hit buffer " << pe.size() << " smaller than " << nHit << " Hits";
   }
   return hit;
 }
 
-bool HFShowerLibrary::rInside(double r) { return (r >= rMin && r <= rMax); }
+bool HFShowerLibrary::rInside(double r) const { return (r >= rMin_ && r <= rMax_); }
+
+HFShowerLibrary::BranchCache::BranchCache(HFShowerLibrary::BranchReader& iReader, size_t maxRecordsToCache) {
+  auto nRecords = iReader.numberOfRecords();
+  if (nRecords > maxRecordsToCache) {
+    nRecords = maxRecordsToCache;
+  }
+  offsets_.reserve(nRecords + 1);
```

Comment:
`offsets_[0+record]` returns the starting element in `photons_` to read while `offsets_[1+record]` gives one passed the end. Hence why we need 1 more element.